src/version/versionpb/version.go

```diff
@@ -0,0 +1,67 @@
+package versionpb
+
+import (
+	fmt "fmt"
+	"strings"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"golang.org/x/mod/semver"
+)
+
+// Canonical returns the version in canonical semver form, according to golang.org/x/mod/semver.
+func (v *Version) Canonical() string {
+	// vMAJOR[.MINOR[.PATCH[-PRERELEASE][+BUILD]]]
+	b := new(strings.Builder)
+	fmt.Fprintf(b, "v%d.%d.%d", v.Major, v.Minor, v.Micro)
+	if additional := v.Additional; additional != "" {
+		// We seem to build the - into the additional field, at least for nightlies.  Trim
+		// it anyway.  We also seem to put + there in some build scripts, which isn't a
+		// format semver can parse.  We map all these to - to fit their spec.
+		additional = strings.TrimLeft(additional, "-.+")
+		b.WriteByte('-')
+		b.WriteString(additional)
+	}
+	return semver.Canonical(b.String())
+}
+
+// IsDevelopment returns true if the version is from a local unreleased build.
+func (v *Version) IsDevelopment() bool {
+	return v.Major == 0 && v.Minor == 0 && v.Micro == 0
+}
+
+var (
+	ErrClientTooOld        = errors.New("client is outdated")
+	ErrServerTooOld        = errors.New("server is outdated")
+	ErrIncompatiblePreview = errors.New("client and server versions must match exactly")
+)
+
+// IsCompatible determines if two versions are compatible.
```

Comment:
Is it worth spelling out the compatibility rules in English here, or is that duplicating the code (and hence liable to get out of date)?

Reply:
Done.